app/uk/gov/hmrc/cataloguefrontend/service/DefaultBranchesService.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cataloguefrontend.service
+
+import uk.gov.hmrc.cataloguefrontend.connector.RepositoryDisplayDetails
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject._
+import scala.concurrent.ExecutionContext
+import scala.language.postfixOps
+
+@Singleton
+class DefaultBranchesService @Inject()()(implicit val ec: ExecutionContext){
+
+  def allTeams(repos: Seq[RepositoryDisplayDetails]): Seq[String] = {
+    val repositories = repos.map(t => t.teamNames)
+    repositories.flatten.distinct.sorted
+  }
+
+  def updatedDefaultBranchCount(
+      repos:            Seq[RepositoryDisplayDetails],
+      name:             Option[String],
+      defaultBranch:    Option[String],
+      teamNames:        Option[String],
+      singleOwnership:  Option[Boolean],
+      archived:         Option[Boolean]): Int = {
+
+    var results: Seq[RepositoryDisplayDetails] = repos
```

Comment:
can we rewrite this to avoid the `var`? Also avoid using `if (opt.isDefined) opt.get` pattern (which can break with refactoring).
Just one idea:
```scala
implicit class Syntax(repos: Seq[RepositoryDisplayDetails]) {
  def filterBy[A](optA: Option[A], f: RepositoryDisplayDetails => A => Boolean): Seq[RepositoryDisplayDetails] =
    optA.fold(repos)(a => repos.filter(repo => f(repo, a))
}
val results =
  repos
    .filterBy(name, _.name contains _)
    .filterBy(singleOwnership)(_ => _.teamNames.length == 1)
```

Reply:
Or better (also with singleOwnership and archived as Boolean rather than a three state Option[Boolean])
```scala
repos
  .filter(repo => optName.fold(true)(repo._name contains _)
  .filter(repo => !singleOwnership || repo.timeNames.length == 1)
  .filter(repo => !onlyNonArchived || !repo.isArchived)
```